components/eam/src/physics/cam/check_energy.F90

```diff
@@ -1228,18 +1228,21 @@ subroutine energy_helper_eam_def_column(u,v,T,q,ps,pdel,phis, &
     end do
     se = se + phis*ps/gravit
 
-    do k = 1, pver
-       wl = wl + q(k,icldliq)*pdel(k)/gravit
-       wi = wi + q(k,icldice)*pdel(k)/gravit
-    end do
+    if (icldliq > 1 .and. icldice > 1) then
+       do k = 1, pver
+          wl = wl + q(k,icldliq)*pdel(k)/gravit
+          wi = wi + q(k,icldice)*pdel(k)/gravit
+       end do
+    end if
 
-    do k = 1, pver
-       wr = wr + q(k,irain)*pdel(k)/gravit
-       ! Recall, P3 does not have a snow constituent.
-       if (.not.microp_scheme == 'P3') then
-         ws = ws + q(k,isnow)*pdel(k)/gravit
-       end if
-    end do
+    if (irain > 1 .and. isnow > 1) then
```

Comment:
I think we will need to break this if up into an `if (irain>1)` and a `if (isnow>1 && microp_scheme!="P3")` set of conditionals.

Reply:
maybe add an "else ws(i) = 0` and then we can remove the `P3` conditional up above.